src/util.js

```diff
@@ -1,12 +1,16 @@
-export function _normalizeAbsolutePath(path, hasPushState) {
+export function _normalizeAbsolutePath(path, hasPushState, absolute = false) {
   if (!hasPushState && path[0] !== '#') {
     path = '#' + path;
   }
 
+  if (hasPushState && absolute) {
+    path = path.substring(1, path.length);
+  }
+
   return path;
 }
 
-export function _createRootedPath(fragment, baseUrl, hasPushState) {
+export function _createRootedPath(fragment, baseUrl, hasPushState, absolute) {
```

Comment:
The hash character should be part of the history's base URL. The router only cares about the path fragment, and the hash is really an implementation detail of the history.